src/junglescout/session.py

```diff
@@ -51,7 +47,92 @@ def login(self, api_key_name: str, api_key: str, api_type: ApiType = ApiType.JS)
 
         Args:
             api_key_name: The name of the API key.
-            api_key: The API
+            api_key: The API key.
             api_type: The type of API to use.
         """
         self.headers.update({"Authorization": f"{api_key_name}:{api_key}", "X_API_Type": api_type.value})
+
+
+class SyncSession(BaseSession):
+    """Represents a synchronous session with the Jungle Scout API."""
+
+    def __init__(self, headers: dict):
+        """Initializes a synchronous session with the given headers.
+
+        Args:
+            headers: A dictionary of HTTP headers to include in requests.
+        """
+        super().__init__(headers)
+
+    def request(self, method: str, url: str, **kwargs) -> httpx.Response:
+        """Sends a request using the synchronous client.
+
+        Args:
+            method: The HTTP method to use (e.g., 'GET', 'POST').
+            url: The URL to send the request to.
+            **kwargs: Additional arguments to pass to the request.
+
+        Returns:
+            httpx.Response: The response from the server.
+        """
+        return self.client.request(method, url, **kwargs)
+
+    def close(self):
+        """Closes the synchronous client session."""
+        self.client.close()
+
+    def login(self, api_key_name: str, api_key: str, api_type: ApiType = ApiType.JS):
+        """Sets the authorization headers for the session.
+
+        Args:
+            api_key_name: The name of the API key.
+            api_key: The API key.
+            api_type: The type of API to use.
+        """
+        super().login(api_key_name, api_key, api_type)
+        self.client = httpx.Client(
+            headers=self.headers,
+            timeout=httpx.Timeout(
+                timeout=self.default_read_timeout,
+                connect=self.default_connect_timeout,
+                read=self.default_read_timeout,
+            ),
+        )
+
+
+class AsyncSession(BaseSession):
+    """Represents an asynchronous session with the Jungle Scout API."""
+
+    def __init__(self, headers: dict):
+        """Initializes an asynchronous session with the given headers.
+
+        Args:
+            headers: A dictionary of HTTP headers to include in requests.
+        """
+        super().__init__(headers)
+
+    async def request(self, method: str, url: str, **kwargs) -> httpx.Response:
+        """Sends a request using the asynchronous client.
+
+        Args:
+            method: The HTTP method to use (e.g., 'GET', 'POST').
+            url: The URL to send the request to.
+            **kwargs: Additional arguments to pass to the request.
+
+        Returns:
+            httpx.Response: The response from the server.
+        """
+        async with self.client as client:
+            request = await self.client.request(method, url, **kwargs)
```

Comment:
`client` defined in the async context manager isn't being used. Should this be:

```python
async with self.client as client:
    request = await client.request(method, url, **kwargs)
```

Reply:
Thanks for catching that!